### PR TITLE
 [diffusion] Extract post-training weight APIs into mixins and add tensor update/checker paths

### PR DESCRIPTION
**Pull request by MikukuOvO**
## Motivation

This PR consolidates diffusion post-training weight management under dedicated mixins and completes the tensor-based update/checker path for diffusion RL/post-training workflows.

Before this change, diffusion post-training weight operations were split across scheduler/worker implementations, and the tensor update / verification flow was not routed through the same post-training mixin structure as the disk update path. This PR makes the post-training API surface more consistent and easier to extend while keeping the runtime behavior explicit.

## Modifications

- Extracted diffusion post-training worker logic into `GPUWorkerPostTrainingMixin`.
- Extracted diffusion post-training scheduler handlers into `SchedulerPostTrainingMixin`.
- Moved the existing diffusion `update_weights_from_disk` and `get_weights_checksum` paths onto the new mixin-based structure.
- Added tensor-based diffusion weight update support through:
  - `UpdateWeightFromTensorReqInput`
  - `POST /update_weights_from_tensor`
  - scheduler dispatch + worker handling for deserializing per-rank tensor payloads
- Added tensor-update verification support through:
  - `UpdateWeightFromTensorCheckerReqInput`
  - `POST /update_weights_from_tensor_checker`
  - `UpdateWeightFromTensorChecker` utility for live transformer verification
- Extended `WeightsUpdater` with:
  - `update_weights_from_tensor`
  - module-scoped payload resolution
  - `flattened_bucket` reconstruction
  - `weight_loader`-aware loading
  - DTensor-aware copy/update handling
- Added TP-aware payload selection in the worker mixin and a TP barrier in the scheduler tensor-update path so success is returned only after all TP ranks finish the update.
- Added SHA-256-based transformer verification logic that:
  - hashes live tensors in a stable way
  - supports DTensor local shards
  - reconstructs/checks TP-sharded tensors on the root rank when needed
  - returns clearer mismatch / missing-tensor diagnostics

## Accuracy Tests

This PR does not modify model forward math or inference kernels. It adds/refactors post-training weight-management and verification paths.

Manual functional validation run in `miles-diffusion`:

- `conda run -n miles-diffusion python /tmp/test_diffusion_post_training_tensor_checker.py`
- `conda run -n miles-diffusion python /tmp/test_diffusion_post_training_server_tensor_checker.py`

Results:

- worker-level `update_weights_from_tensor -> checker`: PASS
- server-level `POST /update_weights_from_tensor`: PASS
- server-level `POST /update_weights_from_tensor_checker` success path: PASS
- server-level `POST /update_weights_from_tensor_checker` mismatch path: PASS (expected failure)
- server-level `POST /get_weights_checksum`: PASS

Example server-level validation on `Tongyi-MAI/Z-Image-Turbo`:

- updated tensor: `noise_refiner.0.ffn_norm1.weight`
- transformer checksum before update:
  - `760694bc3805aac30827f46150f505facb973810599961ffd05aa3a6a2fdaa2e`
- transformer checksum after update:
  - `74da79e1a3225a2f25e17572c987ff14197f0c97d810687bdc54e55cb8f9bfa4`

## Speed Tests and Profiling

N/A.

This PR only affects diffusion post-training weight update / verification paths and does not change the normal inference hot path. No dedicated speed benchmarking or profiling was run.

## Related PRs

This PR builds on the diffusion post-training weight-update work introduced earlier:

- [#18306](https://github.com/sgl-project/sglang/pull/18306) [Feature] Implement `update_weights_from_disk` for SGLang-D
- [#20464](https://github.com/sgl-project/sglang/pull/20464) Add `update_weights_from_tensor` pipeline to Diffusion
- [#21106](https://github.com/sgl-project/sglang/pull/21106) [diffusion] Add `update_weights_from_tensor` checker

If this stack is being split/reviewed incrementally, the local branch also contains these related commits:

- [`6cdd0b542`](https://github.com/sgl-project/sglang/commit/6cdd0b542) `refactor: extract diffusion disk weight update mixins`
- [`5dfbefac3`](https://github.com/sgl-project/sglang/commit/5dfbefac3) `feat: add diffusion tensor weight update mixin path`
- [`75e0cdbb3`](https://github.com/sgl-project/sglang/commit/75e0cdbb3) `refactor: extract diffusion weight checksum mixins`
- [`cbb0f690f`](https://github.com/sgl-project/sglang/commit/cbb0f690f) `feat: add diffusion tensor weight checker mixin path`

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #27181219523](https://github.com/sgl-project/sglang/actions/runs/27181219523)
Latest PR Test (Extra): :x: [Run #27181219466](https://github.com/sgl-project/sglang/actions/runs/27181219466)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/post_training/gpu_worker_post_training_mixin.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    WeightsUpdater,
+    get_updatable_modules,
+)
+from sglang.multimodal_gen.runtime.utils.layerwise_offload import (
+    iter_materialized_weights,
+)
+from sglang.srt.utils import MultiprocessingSerializer
+from sglang.srt.utils.patch_torch import monkey_patch_torch_reductions
+
+if TYPE_CHECKING:
```
Comment (Rockdu) [inline python/sglang/multimodal_gen/runtime/post_training/gpu_worker_post_training_mixin.py:20]: we may need to remove this if we won't have loop import problem

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py [hunk 2]
```diff
@@ -447,9 +447,19 @@ def load_model_from_full_model_state_dict(
                 sharded_tensor = sharded_tensor.to("cpu")
 
         requires_grad = False
-        sharded_sd[target_param_name] = nn.Parameter(
-            sharded_tensor, requires_grad=requires_grad
-        )
+        actual_param = param_dict.get(target_param_name)
```
Comment (Rockdu) [inline python/sglang/multimodal_gen/runtime/loader/fsdp_load.py:450]: Is it necessary to modify fsdp code here?

Comment (gxlvera) [reply python/sglang/multimodal_gen/runtime/loader/fsdp_load.py:450]: Yes. this block of code will run no matter if you switch on fsdp

**Review by mickqian — APPROVED** [review]

**Review by mickqian — APPROVED** [review]
